evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -58,6 +60,23 @@ def _prepare_data(self, X, y):
             pd.DataFrame, pd.Series: Prepared X and y data as pandas types
         """
         X = infer_feature_types(X)
+        if len(X.ww.select(IntegerNullable).columns) > 0:
```

Comment:
small nit, but might be cleaner/faster to set a variable as `X.ww.select(IntNull).columns)` and then use that var in the `if` statement
```python
intNullCols = X.ww.select(IntegerNullable).columns
if len(intNullCols) > 0:
   ...
```